documentation/code.md

```diff
@@ -0,0 +1,231 @@
+# Code documentation guidelines
+
+This document discusses guidelines for writing package and module
+level documentation. For explicit standards on documentation and
+general style, refer to the
+`[style guide](https://github.com/tweag/guides/tree/master/style/Haskell.md)`.
+
+We define code documentation of a package to be the comments in all
+the source files in the package, plus any examples, plus any text
+files explaining the package (e.g. the README).
+
+Code documentation should be useful to both users and maintainers of a
+package. Documentation must:
+
+- Provide an overview of what problem each package and module solves.
+- Explain the strategy used to solve the problem.
+- Explain the details of the implementation where the intention of the
+  author is not readily apparent from reading the code.
+- Point to examples and material explaining how to use a package.
+- Help find the code responsible for performing a specific task.
```

Comment:
I'd rank these above "explain the details of the implementation" and probably also "explain the strategy used to solve the problem" in priority. From a users's perspective I care more about how to use a package than how it is implemented internally.

Reply:
I tried to address this.